### PR TITLE
Added layer_test in addons and refactoring the code

### PR DESCRIPTION
**Pull request by ashutosh1919**
We need to modify code in many files to resolve issue #1567 .  
This PR adds `layer_test` in `testing_utils` of addons.
@gabrieldemarmiesse - I have removed all codes in the function corresponding to model checking and `set|get_weights()`. Currently, tests will fail because the method doesn't return anything (previously it was returning something like `actual_output` which was part of Sequential API tests part.). Please review the code and guide me to the right direction.

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
* We don't need a model for the tests but we still need to do a foward pass to ensure that no exception is raised. To get the output values just do `output_tensor = layer(input_tensor).numpy()`.  So we can still return the result without using a keras model.

* Another thing to check is that the layer output the same numbers after being serialized and deserialized.

**Comment by ashutosh1919** [comment]
@gabrieldemarmiesse , I have added codes as per you comment. But still many tests fails. Tests fails because of lib `NotFoundError`. Please can you take a look and give me a hint?

File: tensorflow_addons/utils/test_utils.py [hunk 1]
```diff
@@ -25,8 +25,90 @@
...
+    # Test adapt, if data was passed.
+    if adapt_data is not None:
+        layer.adapt(adapt_data)
+
+    x = tf.keras.Input(shape=input_shape[1:], dtype=input_dtype)
+    y = layer(x)
+    model = tf.keras.Model(x, y)
+    actual_output = model.predict(input_data)
+    model_config = model.get_config()
+    recovered_model = tf.keras.models.model_from_config(model_config, custom_objects)
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/utils/test_utils.py:99]:
```suggestion
    actual_output = layer(input_data)
    layer_config= tf.keras.layers.serialize(layer)
    recovered_layer = tf.keras.layers.deserialize(layer)
```

Comment (ashutosh1919) [reply tensorflow_addons/utils/test_utils.py]: done changes

**Comment by ashutosh1919** [comment]
cc @bhack

**Comment by seanpmorgan** [comment]
Thank you for your contribution. We sincerely apologize for any delay in reviewing, but TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)

**Pull request closed without merging**